src/driver-sinowealth.c

```diff
@@ -129,13 +131,14 @@ struct sinowealth_config_report {
 	 * 0x2 - 3 mm
 	 */
 	uint8_t lift_off_distance;
-	uint8_t padding[SINOWEALTH_CONFIG_SIZE - SINOWEALTH_CONFIG_SIZE_USED];
+	uint8_t unknown5[36];
+	uint8_t padding[SINOWEALTH_CONFIG_SIZE - SINOWEALTH_CONFIG_SIZE_USED_LONG];
 } __attribute__((packed));
 
 _Static_assert(sizeof(struct sinowealth_config_report) == SINOWEALTH_CONFIG_SIZE, "Invalid size");
 
 struct sinowealth_data {
-	/* this is kinda unnecessary at this time, but all the other drivers do it too ;) */
+	bool is_long;
```

Comment:
It doesn't look pretty with current implementation. Maybe it would make more sense if instead of `is_long` we had `config_size` and `report_id_config` here?